compose.js

```diff
@@ -88,8 +88,12 @@ function mergeComposable(dstDescriptor, srcComposable) {
   combineProperty('configuration', assign);
   combineProperty('deepConfiguration', merge);
   if (Array.isArray(srcDescriptor.initializers)) {
-    if (!Array.isArray(dstDescriptor.initializers)) dstDescriptor.initializers = [];
-    dstDescriptor.initializers.push.apply(dstDescriptor.initializers, srcDescriptor.initializers.filter(isFunction));
+    dstDescriptor.initializers = srcDescriptor.initializers.reduce((result, init) => {
+      if (isFunction(init) && result.indexOf(init) < 0) {
```

Comment:
Using `-1` always felt weird to be as some arbitrary number. You can even reason better about it like this: "index is below 0" instead of "index is equal this weird number".